File: backend/src/main/java/com/leavemarker/entity/Holiday.java
package com.leavemarker.entity;

import com.leavemarker.enums.HolidayType;
import com.leavemarker.enums.IndianState;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Entity
@Table(name = "holidays")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Holiday extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(nullable = false)
    private LocalDate date;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private HolidayType type;

    @Enumerated(EnumType.STRING)
    @Column(length = 50)
    private IndianState state;

    @Column(nullable = false)
    private Boolean active = true;
}

File: backend-go/internal/models/holiday.go
package models

import (
	"time"

	"github.com/leavemarker/backend-go/internal/enums"
)

// Holiday represents a holiday entity
type Holiday struct {
	BaseModel
	CompanyID uint              `gorm:"not null;index" json:"companyId"`
	Name      string            `gorm:"size:200;not null" json:"name"`
	Date      time.Time         `gorm:"type:date;not null" json:"date"`
	Type      enums.HolidayType `gorm:"size:20;not null" json:"type"`
	State     *enums.IndianState `gorm:"size:50" json:"state"`
	Active    bool              `gorm:"default:true" json:"active"`

	// Relationships
	Company Company `gorm:"foreignKey:CompanyID" json:"-"`
}

func (Holiday) TableName() string {
	return "holidays"
}
